docs/weave-tutorial/03-codec.md

```diff
@@ -12,108 +12,118 @@ Weave defines `Marshaller` and `Persistent` interface standards. These interface
 
 ## State
 
-In the previous section, we explained how to define your domain attributes and models. Here `Trade` is defined as proto message. The [x/codec.proto](https://github.com/iov-one/tutorial/blob/master/x/order book/codec.proto#L75-L88) file defines the `Trade` type rather simply, once you remove the comments, this is all that is left:
+In the previous section, we explained how to define your domain attributes and models. Here `Blog` is defined as proto message. The [x/codec.proto](https://orkunkl.github.com/iov-one/blog-tutorial/blob/master/x/blog/codec.proto#L22-L34) file defines the `Blog` type rather simply, once you remove the comments, this is all that is left:
 
 ```protobuf
-// Trade is a settled partial/full order
-// We store these as independent entities to help with queries to map
-// the prices over time. They are also referenced by the Orders, so we can
-// see how much was fulfilled.
-//
-// Ask and Bid tickers are "inherited" from the order book
-message Trade {
+message Blog {
   weave.Metadata metadata = 1;
+  // ID is blog's identifier
   bytes id = 2 [(gogoproto.customname) = "ID"];
-  bytes order_book_id = 3 [(gogoproto.customname) = "OrderBookID"];
-  bytes order_id = 4 [(gogoproto.customname) = "OrderID"];
-  // Address of taker (this is an order that was instantly fulfilled)
-  bytes taker = 5 [(gogoproto.casttype) = "github.com/iov-one/weave.Address"];
-  // Address of maker (this is an order that was stored first before fulfillment)
-  bytes maker = 6 [(gogoproto.casttype) = "github.com/iov-one/weave.Address"];
-  // this is how much each side paid (they got the opposite one)
-  coin.Coin maker_paid = 7;
-  coin.Coin taker_paid = 8;
-  // executed_at defines execution time of an order
-  int64 executed_at = 9 [(gogoproto.casttype) = "github.com/iov-one/weave.UnixTime"];
+  // Owner is the owner address if the blog
+  bytes owner = 3 [(gogoproto.casttype) = "github.com/iov-one/weave.Address"];
+  // Title is title of the blog
+  string title = 4;
+  // Description is description section of the blog
+  string description = 5;
+  // CreatedAt defines creation time of the blog
+  int64 created_at = 6 [(gogoproto.casttype) = "github.com/iov-one/weave.UnixTime"];
 }
 ```
 
-As you can see above, weave is heavily using `bytes` for identities, addresses, etc. At first glance, this might seem difficult to handle but if you take a look at [x/orderbook/bucket](https://github.com/iov-one/tutorial/blob/master/x/orderbook/bucket.go#L125) you can see it is useful for indexing and performance.
+<!---
+TODO refactor here when bucket is merged
+-->
+
+As you can see above, weave is heavily using `bytes` for identities, addresses, etc. At first glance, this might seem difficult to handle but if you take a look at [x/blog/bucket](https://github.com/iov-one/tutorial/blob/master/x/orderbook/bucket.go#L125) you can see it is useful for indexing and performance.
```

Comment:
```suggestion
As you can see above, Weave is heavily using `bytes` for identities, addresses, etc. At first glance, this might seem difficult to handle but if you take a look at [x/blog/bucket](https://github.com/iov-one/tutorial/blob/master/x/orderbook/bucket.go#L125) you can see it is useful for indexing and performance.
```